src/api-service/__app__/onefuzzlib/azure/vmss.py

```diff
@@ -35,39 +37,45 @@ def list_vmss(name: UUID) -> Optional[List[str]]:
             )
         ]
         return instances
-    except CloudError as err:
+    except (ResourceNotFoundError, CloudError) as err:
         logging.error("cloud error listing vmss: %s (%s)", name, err)
 
     return None
 
 
-def delete_vmss(name: UUID) -> Any:
+def delete_vmss(name: UUID) -> bool:
     resource_group = get_base_resource_group()
-    compute_client = mgmt_client_factory(ComputeManagementClient)
+    compute_client = get_compute_client()
     try:
-        compute_client.virtual_machine_scale_sets.delete(resource_group, str(name))
-    except CloudError as err:
+        compute_client.virtual_machine_scale_sets.begin_delete(
+            resource_group, str(name)
+        )
+    except (ResourceNotFoundError, CloudError) as err:
```

Comment:
Review TODO: what since we now catch these errors and map them to a value, what do we do with the resulting `bool`?

Reply:
Based on review of at least one of the use sites for this function, we may want to distinguish the outcomes, or document why `CloudError` and `ResourceNotFoundError` should be treated as equivalent.

Reply:
Addressed (see response in other comment)